packages/frontend/src/features/sqlRunner/components/SaveSqlChartModal.tsx

```diff
@@ -1,40 +1,59 @@
 import { ChartKind } from '@lightdash/common';
-import { Button, Group, Modal, Stack, Text, TextInput } from '@mantine/core';
+import {
+    Button,
+    Group,
+    Modal,
+    Stack,
+    Text,
+    Textarea,
+    TextInput,
+} from '@mantine/core';
 import { useForm, zodResolver } from '@mantine/form';
 import { IconChartBar } from '@tabler/icons-react';
-import { useCallback, useEffect } from 'react';
+import { useCallback, useEffect, type FC } from 'react';
 import { z } from 'zod';
 import MantineIcon from '../../../components/common/MantineIcon';
 import { useSpaceSummaries } from '../../../hooks/useSpaces';
 import { useCreateSqlChartMutation } from '../hooks/useSavedSqlCharts';
 import { useAppDispatch, useAppSelector } from '../store/hooks';
-import { toggleModal } from '../store/sqlRunnerSlice';
+import { updateName } from '../store/sqlRunnerSlice';
 
 const validationSchema = z.object({
     name: z.string().min(1),
+    description: z.string(),
 });
 
 type FormValues = z.infer<typeof validationSchema>;
 
-export const SaveSqlChartModal = () => {
+type Props = {
+    isOpen: boolean;
+    onClose: () => void;
+};
+
+export const SaveSqlChartModal: FC<Props> = ({ isOpen, onClose }) => {
     const dispatch = useAppDispatch();
     const projectUuid = useAppSelector((state) => state.sqlRunner.projectUuid);
     const { data: spaces = [] } = useSpaceSummaries(projectUuid, true);
 
+    const name = useAppSelector((state) => state.sqlRunner.name);
+    const description = useAppSelector((state) => state.sqlRunner.description);
+
     const form = useForm<FormValues>({
         initialValues: {
             name: '',
+            description: '',
```

Comment:
have we thought about having `name` and `description` from RTK in the `initialValues` of this form?
```
 const form = useForm<FormValues>({
        initialValues: {
            name,
            description,
```

Wonder if this would be a good approach 🤔

Reply:
They get put there in the useEffect below. I don't like that approach, but theres a race condition populating this. If you just put them here directly, they don't get set. If you know of a better way, let me know.

Reply:
got it! ignore then :)